File: src/view/MainFrame.java
package view;

import java.awt.EventQueue;

import javax.swing.JFrame;

public class MainFrame {

	private JFrame frame;

	/**
	 * Launch the application.
	 */
	public static void main(String[] args) {
		EventQueue.invokeLater(new Runnable() {
			public void run() {
				try {
					MainFrame window = new MainFrame("Title");
					window.frame.setVisible(true);
				} catch (Exception e) {
					e.printStackTrace();
				}
			}
		});
	}

	/**
	 * Create the application.
	 */
	public MainFrame(String title) {
		initialize(title);
	}

	/**
	 * Initialize the contents of the frame.
	 */
	private void initialize(String title) {
		frame = new JFrame(title);
		frame.setBounds(100, 100, 800, 600);
		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		MainPanel contentPane=new MainPanel();
		frame.setContentPane(contentPane);
	}

}

File: src/demo/som/view/MainFrame.scala

package demo.som.view

import javax.swing._
import java.awt._

import myutils.debug.Debug
import myutils.lang.Worker

object MainFrame {
  /**
   * Launch the application.
   */
  def start(title: String) {
    EventQueue.invokeLater(new Runnable {
      override def run(): Unit = {
        try {
          val window = new MainFrame()
          window.frame.setVisible(true)
          Worker.forkAndStart({
            window.contentPane.contentPanel.somCanvas.start()
          }, 1000)
        } catch {
          case e: Exception => Debug.loge("MainFrame", e)
        }
      }
    })
  }
}

class MainFrame {
  private var frame: JFrame = null
  val contentPane: MainPanel = new MainPanel

  /**
   * Create the application.
   */
  def this(title: String) {
    this()
    initialize(title)
  }

  /**
   * Initialize the contents of the frame.
   */
  private def initialize(title: String) {
    frame = new JFrame(title)
    frame.setBounds(100, 100, 800, 600)
    frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE)
    frame.setContentPane(contentPane)
  }
}